### PR TITLE
#47 Add redux saga wrappers around the API for fetching menu links

### PR DESCRIPTION
**Pull request by justafish**
_Now this is a story all about how
My life got flipped-turned upside down
And I'd like to take a minute
Just sit right there
I'll tell you how I found out that make-cancelable doesn't work_

tl;dr if you read further down [the thread that's linked from make-cancelable](https://github.com/facebook/react/issues/5465), there's an explanation for why it doesn't work.

So, I've added redux-saga because:
- It handles this really well
- We're going to need global state anyway
- It's [v](https://github.com/redux-saga/redux-saga/stargazers) [popular](https://www.npmjs.com/package/redux-saga)
- It's [v testable](https://redux-saga.js.org/docs/introduction/BeginnerTutorial.html#making-our-code-testable)

I also added a nice loading bar

This only partially completes #47, I just ported over the menu code and the Default template so this didn't become un-reviewable.

### CODE REVIEW

File: src/utils/api/api.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  init.credentials = 'include';
+  switch (endpoint) {
+    case 'menu':
+      url = '/admin-api/menu?_format=json';
+      break;
+    default:
+      break;
+  }
+
+  const data = await fetch(
```
Comment (dawehner) [inline src/utils/api/api.js]: This is missing ```{	
         credentials: 'include',	
        },```

Comment (dawehner) [inline src/utils/api/api.js]: Is there a reason for this mix of await and promises?

File: src/reducers/application.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { SET_ERROR, MENU_LOADED } from '../actions/application';
+
+const initialState = {
+  error: null,
```
[the remark this answers is not in the document]
Comment (justafish) [reply src/reducers/application.js]: @dawehner yeah that's a really good idea. Is this something that you think should be done in this PR or can we work on something in a follow up?

File: src/utils/api/api.js (hunk 1 above)
Comment (justafish) [reply src/utils/api/api.js]: @dawehner how else would you write this?

File: src/reducers/application.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (dawehner) [reply src/reducers/application.js]: I'm happy to have a look at this in a follow up.

File: src/utils/api/api.js (hunk 1 above)
Comment (dawehner) [reply src/utils/api/api.js]: Couldn't you use ```const res = await fetch(
  process.env.REACT_APP_DRUPAL_BASE_URL + url,
  init,
);
const data = await res.json();
return data;
```

**Comment by mattgrill** [comment]
@justafish @dawehner needs a rebase/merge with master.

**Pull request merged**